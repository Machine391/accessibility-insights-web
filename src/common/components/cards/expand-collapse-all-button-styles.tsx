// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
// .expand-collapse-all-button {
//     display: flex;
//     padding: unset;
//     padding-right: 27px !important;
//     padding-left: 1px !important;
//     margin-left: 1px !important;
//     margin-top: 2px !important;
//     font-weight: ;

import { makeStyles, tokens } from "@fluentui/react-components";


// }

export const useExpandCollapseAllButtonStyles = makeStyles({
    expandCollapseAllButton: {
        display: 'flex',
        padding: 'unset',
        paddingRight: '27px!important',
        paddingLeft: '1px!important',
        marginLeft: '1px!important',
        marginTop: '2px!important',
        fontWeight: tokens?.fontSizeBase400
    }
})
