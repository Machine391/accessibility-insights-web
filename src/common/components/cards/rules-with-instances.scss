// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../../common/styles/colors.scss';
@import '../../../common/styles/fonts.scss';

.rule-details-group {
    .rule-detail {
        font-size: 14px;
        padding-top: 16px;
        padding-right: 8px;
        padding-bottom: 16px;
        padding-left: 8px;
        display: flex;
        align-items: baseline;
        text-align: left;
        font-weight: normal;

        .outcome-chip {
            vertical-align: middle;
            margin-bottom: 2px;
        }

        .rule-details-id {
            font-weight: $font-weight-semi-bold;
            color: $primary-text;
            word-break: break-all;

            a {
                font-weight: $font-weight-semi-bold;
                color: $primary-text !important;
            }

            strong {
                font-weight: $font-weight-semi-bold;
            }
        }

        .rule-detail-description {
            color: $secondary-text !important;
            word-break: break-all;
        }
    }
}

.collapsible-rule-details-group {
    box-shadow: 0 1px 0 rgb(0 0 0 / 8%);

    .rule-detail {
        box-shadow: unset;
    }

    &:not(.collapsed) {
        box-shadow: unset;
    }
}
