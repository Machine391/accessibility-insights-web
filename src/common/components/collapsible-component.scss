// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../common/styles/common.scss';

span.collapsible-title {
    padding-left: 0.5vw;
}

.collapsible {
    display: inline-flex;
    font-size: 26px;
    height: auto;
    width: auto;
    text-align: left;
    color: #106ebe;

    h2 {
        font-weight: normal;
        font-size: 17px;
    }
}

.collapsible-content {
    overflow: hidden;
    padding-left: calc(0.5vw + 13px); // 13px is Chevron width.
}
