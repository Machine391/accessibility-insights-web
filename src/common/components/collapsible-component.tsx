// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ActionButton } from '@fluentui/react';
import { Button, themeToTokensObject, webLightTheme } from '@fluentui/react-components';
import { css } from '@fluentui/utilities';
import styles from 'common/components/collapsible-component.scss';
import * as React from 'react';
import { ChevronDown24Regular, ChevronRight24Regular } from '@fluentui/react-icons';

export interface CollapsibleComponentProps {
    header: JSX.Element;
    content: JSX.Element;
    contentClassName?: string;
    containerClassName?: string;
}

interface CollapsibleComponentState {
    showContent: boolean;
}

export class CollapsibleComponent extends React.Component<
    CollapsibleComponentProps,
    CollapsibleComponentState
> {
    private readonly iconNameDown = 'ChevronDown';
    private readonly iconNameUp = 'ChevronRight';

    constructor(props: CollapsibleComponentProps) {
        super(props);
        this.state = { showContent: true };
    }

    private onClick = (): void => {
        this.setState(prevState => ({ showContent: !prevState.showContent }));
    };

    public render(): JSX.Element {
        const showContent = this.state.showContent;
        let iconName = this.iconNameUp;
        let content: JSX.Element | null = null;

        if (showContent) {
            iconName = this.iconNameDown;
            content = (
                <div className={css(this.props.contentClassName, styles.collapsibleContent)}>
                    {this.props.content}
                </div>
            );
        }

        return (
            <div className={css(this.props.containerClassName)}>
                {/* 111<ActionButton
                    className={styles.collapsible}
                    iconProps={{ iconName: iconName }}
                    onClick={this.onClick}
                    aria-expanded={showContent}
                >
                    {this.props.header}
                </ActionButton> */}
                <Button className={styles.collapsible} appearance='transparent'
                    onClick={this.onClick}
                    aria-expanded={showContent}>
                    {showContent ? (
                        <ChevronDown24Regular color={themeToTokensObject(webLightTheme).colorCompoundBrandStrokeHover} className={styles.collapsible} />
                    ) : (
                        <ChevronRight24Regular color={themeToTokensObject(webLightTheme).colorCompoundBrandStrokeHover} className={styles.collapsible} />
                    )}
                    111
                    {this.props.header}
                </Button>
                {content}
            </div>
        );
    }
}
