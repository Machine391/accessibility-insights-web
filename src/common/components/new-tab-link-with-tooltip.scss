// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/styles/colors.scss';

:global(.high-contrast-theme) {
    .insights-tooltip {
        outline-style: solid;
        outline-width: 1px;
        outline-color: $always-white;
    }
}

.insights-tooltip-host {
    display: inline-block;
    min-width: 0;
}

.insights-link {
    display: flex;

    &:hover,
    &:active {
        text-decoration: none !important;
    }

    &:focus-visible {
        box-shadow: $neutral-60 0 0 0 1px inset !important;
        outline: $neutral-60 auto 1px !important;
        text-decoration: none !important;
    }
}
