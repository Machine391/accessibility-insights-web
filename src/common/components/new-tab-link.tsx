// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Link, LinkProps, mergeClasses } from '@fluentui/react-components';
import * as React from 'react';

import { NamedFC } from '../react/named-fc';

export type NewTabLinkProps = LinkProps & { as?: 'a' };

export const NewTabLink = NamedFC<NewTabLinkProps>('NewTabLink', ({ className, ...props }) => {
    return <Link className={mergeClasses('insights-link', className)} target="_blank" {...props} />;
});
