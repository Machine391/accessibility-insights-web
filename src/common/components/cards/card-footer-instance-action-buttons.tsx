// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { registerIcons } from '@fluentui/react/lib/Styling';
import {
    Button,
    Menu,
    MenuButton,
    MenuItem,
    MenuList,
    MenuPopover,
    MenuTrigger,
    Tooltip,
} from '@fluentui/react-components';
import {
    CardFooterMenuItemsBuilder,
    CardFooterMenuItemsDeps,
} from 'common/components/cards/card-footer-menu-items-builder';
import { CardsViewController } from 'common/components/cards/cards-view-controller';
import { FluentUIV9Icon } from 'common/icons/fluentui-v9-icons';
import { MoreActionsMenuIcon } from 'common/icons/more-actions-menu-icon';
import { NarrowModeStatus } from 'DetailsView/components/narrow-mode-detector';
import * as React from 'react';
import { CreateIssueDetailsTextData } from '../../types/create-issue-details-text-data';
import { UserConfigurationStoreData } from '../../types/store-data/user-configuration-store';
import { Toast, ToastDeps } from '../toast';
import styles from './card-footer-instance-action-buttons.scss';
import { CardInteractionSupport } from './card-interaction-support';

registerIcons({
    icons: {
        MoreActionsMenuIcon: <MoreActionsMenuIcon />,
    },
});

export type CardFooterInstanceActionButtonsDeps = {
    cardInteractionSupport: CardInteractionSupport;
    cardFooterMenuItemsBuilder: CardFooterMenuItemsBuilder;
    cardsViewController: CardsViewController;
} & ToastDeps &
    CardFooterMenuItemsDeps;

export interface CardFooterInstanceActionButtonsProps {
    deps: CardFooterInstanceActionButtonsDeps;
    userConfigurationStoreData: UserConfigurationStoreData | null;
    issueDetailsData: CreateIssueDetailsTextData;
    kebabMenuAriaLabel?: string;
    narrowModeStatus?: NarrowModeStatus;
}

export const CardFooterInstanceActionButtons = props => {
    const toastRef = React.useRef(null);
    const fileIssueButtonRef = React.useRef(null);
    const kebabButtonRef = React.useRef(null);

    const focusButtonAfterDialogClosed = (): void => {
        if (props?.narrowModeStatus?.isCardFooterCollapsed) {
            kebabButtonRef?.current?.focus();
        } else {
            fileIssueButtonRef?.current?.removeAttribute('textprediction');
            fileIssueButtonRef?.current?.setAttribute('writingsuggestions', 'false');
            fileIssueButtonRef?.current?.focus();
        }
    };

    const getMenuItems = () => {
        return props.deps.cardFooterMenuItemsBuilder.getCardFooterMenuItems({
            ...props,
            toastRef: toastRef,
            fileIssueButtonRef: ref => (fileIssueButtonRef.current = ref),
            onIssueFilingSettingsDialogDismissed: focusButtonAfterDialogClosed,
        });
    };

    const renderCopyFailureDetailsToast = () => {
        const { cardInteractionSupport } = props.deps;

        if (!cardInteractionSupport.supportsCopyFailureDetails) {
            return null;
        }

        return <Toast ref={toastRef} deps={props.deps} />;
    };

    const renderKebabButton = () => {
        return (
            <Menu>
                <MenuTrigger>
                    <MenuButton
                        className={styles.menuButton}
                        ref={ref => (kebabButtonRef.current = ref)}
                        appearance="transparent"
                        aria-label={props?.kebabMenuAriaLabel}
                        icon={<FluentUIV9Icon iconName="MoreVerticalRegular" />}
                    />
                </MenuTrigger>
                <MenuPopover
                    style={{
                        padding: 'unset !important',
                        border: 'unset !important',
                        borderRadius: 'unset !important',
                    }}
                >
                    <MenuList>
                        {getMenuItems().map((item: any, index: number) => (
                            <MenuItem
                                className={styles.kebabMenuIcon}
                                key={`${item.key}-${index}-kebabMenuItem`}
                                icon={<FluentUIV9Icon iconName={item?.iconName} />}
                                {...item}
                            >
                                {item?.text}
                            </MenuItem>
                        ))}
                    </MenuList>
                </MenuPopover>
            </Menu>
        );
    };

    const renderExpandedButtons = () => {
        const menuItems = getMenuItems();
        console.log('inside expanded button');
        return (
            <>
                {menuItems.map(props => (
                    <Button
                        as="button"
                        appearance="transparent"
                        onClick={props.onClick}
                        icon={<FluentUIV9Icon iconName={props?.iconName} />}
                        className={styles[props.key]}
                        size="medium"
                        ref={ref => (props.componentRef = ref)}
                        key={`${props.key}-expandedButtons`}
                    >
                        {props.text}
                    </Button>
                ))}
            </>
        );
    };

    const renderButtons = () => {
        if (props.narrowModeStatus?.isCardFooterCollapsed) {
            return renderKebabButton();
        } else {
            return renderExpandedButtons();
        }
    };

    const menuItems = getMenuItems();
    console.log('here--->', menuItems?.length)
    const menuItemsJsx =
        menuItems?.length == 0 ? null : (
            <div onKeyDown={event => event.stopPropagation()}>
                {renderButtons()}
                {renderCopyFailureDetailsToast()}
            </div>
        );

    return menuItemsJsx;
};
