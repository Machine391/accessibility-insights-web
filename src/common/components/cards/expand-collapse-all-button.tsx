// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ActionButton } from '@fluentui/react';
import { CardSelectionMessageCreator } from 'common/message-creators/card-selection-message-creator';
import { NamedFC } from 'common/react/named-fc';
import * as React from 'react';
//import styles from './expand-collapse-all-button.scss';
import { Button, themeToTokensObject, webLightTheme } from '@fluentui/react-components';
import { ChevronDown32Regular, ChevronRight32Regular } from '@fluentui/react-icons';
import { useExpandCollapseAllButtonStyles } from 'common/components/cards/expand-collapse-all-button-styles';

export type ExpandCollapseAllButtonProps = {
    allCardsCollapsed: boolean;
    cardSelectionMessageCreator: CardSelectionMessageCreator;
};

export const ExpandCollapseAllButton = NamedFC<ExpandCollapseAllButtonProps>(
    'ExpandCollapseAllButton',
    props => {
        const getStyles = useExpandCollapseAllButtonStyles()
        const { allCardsCollapsed, cardSelectionMessageCreator } = props;

        let expandCollapseAllButtonHandler = cardSelectionMessageCreator.collapseAllRules;
        let buttonText = 'Collapse all';
        let iconName = <ChevronDown32Regular color={themeToTokensObject(webLightTheme)?.colorCompoundBrandStrokeHover} />;
        let ariaLabel: string | undefined = undefined;

        if (allCardsCollapsed) {
            expandCollapseAllButtonHandler = cardSelectionMessageCreator.expandAllRules;
            buttonText = 'Expand all';
            iconName = <ChevronRight32Regular color={themeToTokensObject(webLightTheme)?.colorCompoundBrandStrokeHover} />;
            ariaLabel = 'Expand all rules to show failed instances.';
        }

        return (
            // <ActionButton
            //     iconProps={{ iconName }}
            //     onClick={expandCollapseAllButtonHandler}
            //     aria-expanded={!allCardsCollapsed}
            //     aria-label={ariaLabel}
            //     className={styles.expandCollapseAllButton}
            // >
            //     <h3>555</h3>
            //     {buttonText}
            // </ActionButton>
            <Button size='medium'
                appearance="transparent"
                className={getStyles?.expandCollapseAllButton}
                aria-label={ariaLabel}
                aria-expanded={!allCardsCollapsed}
                onClick={expandCollapseAllButtonHandler}
                icon={iconName}>
                {buttonText}
            </Button>

        );
    },
);
