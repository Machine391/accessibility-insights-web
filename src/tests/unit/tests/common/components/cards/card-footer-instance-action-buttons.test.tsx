// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Button, Menu, MenuItem, MenuList, MenuPopover, MenuTrigger, Tooltip } from '@fluentui/react-components';
import { render, RenderResult, screen } from '@testing-library/react';
import {
    CardFooterInstanceActionButtons,
    CardFooterInstanceActionButtonsDeps,
    CardFooterInstanceActionButtonsProps,
} from 'common/components/cards/card-footer-instance-action-buttons';
import {
    CardFooterMenuItem,
    CardFooterMenuItemsBuilder,
    CardFooterMenuItemsProps,
} from 'common/components/cards/card-footer-menu-items-builder';
import { allCardInteractionsSupported } from 'common/components/cards/card-interaction-support';
import { CardsViewController } from 'common/components/cards/cards-view-controller';
import { CreateIssueDetailsTextData } from 'common/types/create-issue-details-text-data';
import { UserConfigurationStoreData } from 'common/types/store-data/user-configuration-store';
import { NarrowModeStatus } from 'DetailsView/components/narrow-mode-detector';
import * as React from 'react';
import { IMock, It, Mock, Times } from 'typemoq';
import { Toast } from '../../../../../../common/components/toast';
import {
    expectMockedComponentPropsToMatchSnapshots,
    getMockComponentClassPropsForCall,
    mockReactComponents,
    useOriginalReactElements,
} from '../../../../mock-helpers/mock-module-helpers';
import { IButton } from '@fluentui/react';
import { setAttribute } from '@fluentui/react/lib/components/KeytipData/useKeytipRef';

jest.mock('@fluentui/react-components');
jest.mock('../../../../../../common/components/toast');
describe(CardFooterInstanceActionButtons, () => {
    mockReactComponents([Toast, Button, MenuItem, Menu, MenuTrigger, MenuPopover, MenuList, Tooltip]);

    let defaultProps: CardFooterInstanceActionButtonsProps;
    let defaultDeps: CardFooterInstanceActionButtonsDeps;
    let menuItemsBuilderMock: IMock<CardFooterMenuItemsBuilder>;
    let cardsViewControllerMock: IMock<CardsViewController>;

    const menuItems: CardFooterMenuItem[] = [
        {
            key: 'item1',
            text: 'item 1',
            onClick: () => null,
        },
        {
            key: 'item2',
            text: 'item 2',
            onClick: () => null,
        },
    ];

    beforeEach(() => {
        menuItemsBuilderMock = Mock.ofType<CardFooterMenuItemsBuilder>();
        cardsViewControllerMock = Mock.ofType<CardsViewController>();

        defaultDeps = {
            cardInteractionSupport: allCardInteractionsSupported,
            cardFooterMenuItemsBuilder: menuItemsBuilderMock.object,
            cardsViewController: cardsViewControllerMock.object,
        } as CardFooterInstanceActionButtonsDeps;

        defaultProps = {
            deps: defaultDeps,
            userConfigurationStoreData: {} as UserConfigurationStoreData,
            issueDetailsData: {} as CreateIssueDetailsTextData,
        } as CardFooterInstanceActionButtonsProps;
    });

    it('renders as null with no menu items', () => {
        setupGetMenuItems([], defaultProps);

        const renderResult = render(<CardFooterInstanceActionButtons {...defaultProps} />);
        // renderResult.debug()
        expect(renderResult.container.firstChild).toBeNull();
    });

    it('renders without copyFailureDetails supported', () => {
        const props = {
            ...defaultProps,
            deps: {
                ...defaultDeps,
                cardInteractionSupport: {
                    ...allCardInteractionsSupported,
                    supportsCopyFailureDetails: false,
                },
            },
        };
        setupGetMenuItems(menuItems, props);

        const renderResult = render(<CardFooterInstanceActionButtons {...props} />);

        expect(renderResult.asFragment()).toMatchSnapshot('component snapshot');
    });

    describe.each([true, false])('with isCardFooterCollapsed=%s', isCardFooterCollapsed => {
        beforeEach(() => {
            defaultProps.narrowModeStatus = { isCardFooterCollapsed } as NarrowModeStatus;
        });

        it('renders per snapshot', () => {
            setupGetMenuItems(menuItems, defaultProps);

            const renderResult = render(<CardFooterInstanceActionButtons {...defaultProps} />);

            verifySnapshots(renderResult);
        });

        function verifySnapshots(renderedElement: RenderResult): void {
            useOriginalReactElements('@fluentui/react-components', ['Tooltip', 'Menu', "MenuTrigger", "MenuButton", "MenuPopover", "MenuList", "MenuItem"])
            expect(renderedElement.asFragment()).toMatchSnapshot('component snapshot');
            if (isCardFooterCollapsed) {
                //expectMockedComponentPropsToMatchSnapshots([Button], 'action button menu props');
                expect(renderedElement.asFragment()).toMatchSnapshot('kebab button menu');
            }
        }
    });

    it.each(['test-kebabmenuarialabel', undefined])(
        'renders per snapshot with %s aria label passed as prop',
        ariaLabel => {
            useOriginalReactElements('@fluentui/react-components', ['Tooltip', 'Menu', 'MenuTrigger', 'MenuButton']);
            defaultProps.narrowModeStatus = { isCardFooterCollapsed: true } as NarrowModeStatus;

            const newProps = {
                ...defaultProps,
                kebabMenuAriaLabel: ariaLabel,
            };
            setupGetMenuItems(menuItems, newProps);

            const renderResult = render(<CardFooterInstanceActionButtons {...newProps} />);

            expect(renderResult.asFragment()).toMatchSnapshot('component snapshot');
            // expectMockedComponentPropsToMatchSnapshots([Button], 'action button menu props');
        },
    );

    it("onIssueFilingSettingsDialogDismissed focus kebab button", async () => {
        useOriginalReactElements('@fluentui/react-components', ['Tooltip', 'Menu', 'MenuTrigger', 'MenuButton']);
        defaultProps.narrowModeStatus = { isCardFooterCollapsed: true } as NarrowModeStatus;
        let menuItemsProps: CardFooterMenuItemsProps;

        menuItemsBuilderMock
            .setup(m =>
                m.getCardFooterMenuItems(
                    It.isObjectWith(defaultProps as unknown as CardFooterMenuItemsProps),
                ),
            )
            .returns((props: CardFooterMenuItemsProps) => {
                menuItemsProps = props;
                return menuItems;
            });
        const result = render(<CardFooterInstanceActionButtons {...defaultProps} />);

        // const buttonRef = jest.spyOn(React, 'useRef').mockReturnValue({ current: { focus: jest.fn() } })
        menuItemsProps.onIssueFilingSettingsDialogDismissed();
        const getButton = result.getByRole('button');
        // console.log('here-->', document.activeElement)
        // //expect(buttonRef).toHaveBeenCalled()
        expect(document.activeElement).toBe(getButton);
    })

    // it('onIssueFilingSettingsDialogDismissed focuses kebab button', async () => {
    //     useOriginalReactElements('@fluentui/react-components', ['Button', 'Tooltip'])
    //     defaultProps.narrowModeStatus = { isCardFooterCollapsed: false } as NarrowModeStatus;

    //     const kebabButtonMock = Mock.ofType<IButton>();
    //     let menuItemsProps: CardFooterMenuItemsProps;


    //     menuItemsBuilderMock
    //         .setup(m =>
    //             m.getCardFooterMenuItems(
    //                 It.isObjectWith(defaultProps as unknown as CardFooterMenuItemsProps),
    //             ),
    //         )
    //         .returns((props: CardFooterMenuItemsProps) => {
    //             menuItemsProps = props;
    //             return menuItems;
    //         });

    //     const result = render(<CardFooterInstanceActionButtons {...defaultProps} />);
    //     console.log('result.container[0]', result.container[0]);
    //     const fileIssueButtonRef = jest.spyOn(React, 'useRef')
    //     jest.spyOn(React, 'useRef').mockReturnValue({ current: { focus: jest.fn(), removeAttribute: jest.fn(), setAttribute: jest.fn() } })
    //     //const getButton = result.container[0];
    //     //console.log('here-->', getMockComponentClassPropsForCall(Button))
    //     // call ref callback to set rendered component's ref to our button mock
    //     // const kebabButtonRefCallback = getMockComponentClassPropsForCall(Button).ref as (
    //     //     ref: IButton,
    //     // ) => void;
    //     // kebabButtonRefCallback(kebabButtonMock.object);

    //     menuItemsProps.onIssueFilingSettingsDialogDismissed();
    //     expect(getButton).toHaveFocus();

    //     // kebabButtonMock.verify(k => k.focus(), Times.once());
    // });

    it('File issue button is focused when issue filing settings dialog is dismissed', async () => {
        useOriginalReactElements('@fluentui/react-components', ['Tooltip', 'Button']);
        defaultProps.narrowModeStatus = { isCardFooterCollapsed: false } as NarrowModeStatus;
        //= jest.spyOn(React, 'useRef').mockReturnValue(())
        //  const fileIssueButtonMock = jest.spyOn(React, 'useRef').mockReturnValue({ current: { focus: jest.fn(), removeAttribute: jest.fn(), setAttribute: jest.fn() } })
        //const fileIssueButtonMock = React.useRef<HTMLInputElement>(null)
        let menuItemsProps: CardFooterMenuItemsProps;
        menuItemsBuilderMock
            .setup(m =>
                m.getCardFooterMenuItems(
                    It.isObjectWith(defaultProps as unknown as CardFooterMenuItemsProps),
                ),
            )
            .returns(props => {
                menuItemsProps = props;
                return menuItems;
            });

        const renderResult = render(
            <CardFooterInstanceActionButtons
                {...defaultProps}
                deps={{ ...defaultDeps, cardInteractionSupport: allCardInteractionsSupported }}
            />,
        );


        // call ref callback to set rendered component's ref to our button mock
        //const buttonRefCallback = menuItemsProps.fileIssueButtonRef as any;
        //buttonRefCallback(fileIssueButtonMock);
        //console.log('here-->', buttonRefCallback(fileIssueButtonMock))

        menuItemsProps.onIssueFilingSettingsDialogDismissed();
        renderResult.debug()
        // const getButton = screen.getByText('item 1', { selector: 'button' });
        const getButton = screen.getByText(/item 1/)

        //fileIssueButtonMock.verify(f => f.focus(), Times.once());
        expect(document.activeElement).toBe(getButton);
    });

    function setupGetMenuItems(
        menuItems: CardFooterMenuItem[],
        expectedProps: CardFooterInstanceActionButtonsProps,
    ): void {
        menuItemsBuilderMock
            .setup(m =>
                m.getCardFooterMenuItems(
                    It.isObjectWith(expectedProps as unknown as CardFooterMenuItemsProps),
                ),
            )
            .returns(() => menuItems);
    }
});
