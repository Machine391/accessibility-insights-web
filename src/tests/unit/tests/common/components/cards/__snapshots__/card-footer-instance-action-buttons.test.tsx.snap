// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooterInstanceActionButtons renders per snapshot with test-kebabmenuarialabel aria label passed as prop: component snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      class="fui-Button r1alrhcs fui-MenuButton menuButton ___cpe5h10_1yhfyj7 f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f ft1hn21 fuxngvv fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1dfjoow f1j98vj9 f1tme0vf f4xjyn1 f18onu3q f9ddjv3 f18ktai2 fwbmr0d f44c6la"
      data-tabster="{"restorer":{"type":1}}"
      id="menur1"
      role="button"
      tabindex="0"
      type="button"
    >
      <span
        class="fui-Button__icon rywnvv2 fui-MenuButton__icon"
      >
        <svg
          aria-hidden="true"
          class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
          fill="currentColor"
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 6a1.25 1.25 0 1 1 0-2.5A1.25 1.25 0 0 1 10 6Zm0 5.25a1.25 1.25 0 1 1 0-2.5 1.25 1.25 0 0 1 0 2.5Zm-1.25 4a1.25 1.25 0 1 0 2.5 0 1.25 1.25 0 0 0-2.5 0Z"
            fill="currentColor"
          />
        </svg>
      </span>
    </button>
    <span
      hidden=""
    />
    <mock-toast
      deps="[object Object]"
      timeoutlength="6000"
    />
  </div>
</DocumentFragment>
`;

exports[`CardFooterInstanceActionButtons renders per snapshot with undefined aria label passed as prop: component snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      class="fui-Button r1alrhcs fui-MenuButton menuButton ___cpe5h10_1yhfyj7 f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f ft1hn21 fuxngvv fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1dfjoow f1j98vj9 f1tme0vf f4xjyn1 f18onu3q f9ddjv3 f18ktai2 fwbmr0d f44c6la"
      data-tabster="{"restorer":{"type":1}}"
      id="menur3"
      role="button"
      tabindex="0"
      type="button"
    >
      <span
        class="fui-Button__icon rywnvv2 fui-MenuButton__icon"
      >
        <svg
          aria-hidden="true"
          class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
          fill="currentColor"
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 6a1.25 1.25 0 1 1 0-2.5A1.25 1.25 0 0 1 10 6Zm0 5.25a1.25 1.25 0 1 1 0-2.5 1.25 1.25 0 0 1 0 2.5Zm-1.25 4a1.25 1.25 0 1 0 2.5 0 1.25 1.25 0 0 0-2.5 0Z"
            fill="currentColor"
          />
        </svg>
      </span>
    </button>
    <span
      hidden=""
    />
    <mock-toast
      deps="[object Object]"
      timeoutlength="6000"
    />
  </div>
</DocumentFragment>
`;

exports[`CardFooterInstanceActionButtons renders without copyFailureDetails supported: component snapshot 1`] = `
<DocumentFragment>
  <div>
    <mock-button
      appearance="transparent"
      as="button"
      classname="item1"
      size="medium"
    >
      item 1
    </mock-button>
    <mock-button
      appearance="transparent"
      as="button"
      classname="item2"
      size="medium"
    >
      item 2
    </mock-button>
  </div>
</DocumentFragment>
`;

exports[`CardFooterInstanceActionButtons with isCardFooterCollapsed=false renders per snapshot: component snapshot 1`] = `
<DocumentFragment>
  <div>
    <mock-button
      appearance="transparent"
      as="button"
      classname="item1"
      size="medium"
    >
      item 1
    </mock-button>
    <mock-button
      appearance="transparent"
      as="button"
      classname="item2"
      size="medium"
    >
      item 2
    </mock-button>
    <mock-toast
      deps="[object Object]"
      timeoutlength="6000"
    />
  </div>
</DocumentFragment>
`;

exports[`CardFooterInstanceActionButtons with isCardFooterCollapsed=true renders per snapshot: component snapshot 1`] = `
<DocumentFragment>
  <div>
    <mock-tooltip
      content="More actions"
      relationship="description"
    >
      <mock-menu>
        <mock-menutrigger />
        <mock-menupopover
          style="border-radius: unset !important;"
        >
          <mock-menulist>
            <mock-menuitem
              classname="kebabMenuIcon"
              text="item 1"
            >
              item 1
            </mock-menuitem>
            <mock-menuitem
              classname="kebabMenuIcon"
              text="item 2"
            >
              item 2
            </mock-menuitem>
          </mock-menulist>
        </mock-menupopover>
      </mock-menu>
    </mock-tooltip>
    <mock-toast
      deps="[object Object]"
      timeoutlength="6000"
    />
  </div>
</DocumentFragment>
`;

exports[`CardFooterInstanceActionButtons with isCardFooterCollapsed=true renders per snapshot: kebab button menu 1`] = `
<DocumentFragment>
  <div>
    <mock-tooltip
      content="More actions"
      relationship="description"
    >
      <mock-menu>
        <mock-menutrigger />
        <mock-menupopover
          style="border-radius: unset !important;"
        >
          <mock-menulist>
            <mock-menuitem
              classname="kebabMenuIcon"
              text="item 1"
            >
              item 1
            </mock-menuitem>
            <mock-menuitem
              classname="kebabMenuIcon"
              text="item 2"
            >
              item 2
            </mock-menuitem>
          </mock-menulist>
        </mock-menupopover>
      </mock-menu>
    </mock-tooltip>
    <mock-toast
      deps="[object Object]"
      timeoutlength="6000"
    />
  </div>
</DocumentFragment>
`;
